### PR TITLE
Bump Dynamo to 3.1

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

Bump master to 3.1

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section**

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 1]
```diff
@@ -54,14 +54,12 @@
 // AssemblyVersionAttribute so that the "Product version" in the file
 // properties window matches the version displayed in the GAC shell extension.
 //[assembly: AssemblyInformationalVersion("1.0.0.0")] // a.k.a. "Product version"
-// Version information for an assembly consists of the following four values:
-//
-//      Major Version
-//      Minor Version
+// Version information for an assembly consists of the following four valu
+//      Majo//Minor Version
```
Comment (QilongTang) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs:58]: typo?

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt [hunk 2]
```diff
@@ -68,7 +66,7 @@ using System.Runtime.InteropServices;
 [assembly: AssemblyFileVersion("<#= this.MajorVersion #>.<#= this.MinorVersion #>.<#= this.BuildNumber #>.<#= this.RevisionNumber #>")]
 <#+
 int MajorVersion = 3;
-int MinorVersion = 0;
+int MinorVersion = 1;
```
Comment (QilongTang) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt:69]: Can you update the year in the `RevisionNumber` formula as well to use 2024?

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt [hunk 3]
```diff
@@ -55,10 +55,8 @@ using System.Runtime.InteropServices;
 // AssemblyVersionAttribute so that the "Product version" in the file
 // properties window matches the version displayed in the GAC shell extension.
 //[assembly: AssemblyInformationalVersion("1.0.0.0")] // a.k.a. "Product version"
-// Version information for an assembly consists of the following four values:
-//
-//      Major Version
-//      Minor Version
+// Version information for an assembly consists of the following four valu
+//      Majo//Minor Version
```
[the remark this answers is not in the document]
Comment (sm6srw) [reply src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt:59]: Oops, thanks!

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt (hunk 2 above)
Comment (sm6srw) [reply src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt:69]: Goto 2024 will give negative numbers but we can do 2023 now and then try to remember to bump it again when we get to 3.2.

Comment (QilongTang) [reply src/AssemblySharedInfoGenerator/AssemblySharedInfo.tt:71]: Sure, thanks!

**Pull request merged**